crates/compiler/gen_dev/src/generic64/mod.rs

```diff
@@ -1531,38 +1437,178 @@ impl<
-            }
-            Layout::F64 | Layout::F32 => {
-                let width = if *arg_layout == Layout::F64 {
-                    FloatWidth::F64
+        self.compare(
+            CompareOperation::GreaterThanOrEqual,
+            dst,
+            src1,
+            src2,
+            arg_layout,
+        )
+    }
+
+    fn build_higher_order_lowlevel(
+        &mut self,
+        dst: &Symbol,
+        higher_order: &HigherOrderLowLevel<'a>,
+        ret_layout: InLayout<'a>,
+    ) {
+        let ident_ids = self
+            .interns
+            .all_ident_ids
+            .get_mut(&self.env.module_id)
+            .unwrap();
+
+        let (inc_n_data_symbol, inc_n_data_linker_data) = self.helper_proc_gen.gen_refcount_proc(
+            ident_ids,
+            self.layout_interner,
+            Layout::UNIT,
+            HelperOp::Inc,
+        );
+
+        let caller_proc = CallerProc::new(
+            self.env.arena,
+            self.env.module_id,
+            ident_ids,
+            self.layout_interner,
+            &higher_order.passed_function,
+            higher_order.closure_env_layout,
+        );
+
+        match higher_order.op {
+            HigherOrder::ListMap { xs } => {
+                let old_element_layout = higher_order.passed_function.argument_layouts[0];
+                let new_element_layout = higher_order.passed_function.return_layout;
+
+                let input_list_layout = Layout::Builtin(Builtin::List(old_element_layout));
+                let input_list_in_layout = self.layout_interner.insert(input_list_layout);
+
+                let caller = self.debug_symbol("caller");
+                let data = self.debug_symbol("data");
+                let alignment = self.debug_symbol("alignment");
+                let old_element_width = self.debug_symbol("old_element_width");
+                let new_element_width = self.debug_symbol("new_element_width");
+
+                self.load_layout_alignment(new_element_layout, alignment);
+
+                self.load_layout_stack_size(old_element_layout, old_element_width);
+                self.load_layout_stack_size(new_element_layout, new_element_width);
+
+                self.helper_proc_symbols.extend(inc_n_data_linker_data);
+                self.helper_proc_symbols
+                    .extend([(caller_proc.proc_symbol, caller_proc.proc_layout)]);
+
+                let inc_n_data_string = self.function_symbol_to_string(
+                    inc_n_data_symbol,
+                    std::iter::empty(),
+                    None,
+                    Layout::UNIT,
+                );
+
+                let caller_string = self.function_symbol_to_string(
+                    caller_proc.proc_symbol,
+                    std::iter::empty(),
+                    None,
+                    Layout::UNIT,
+                );
+
+                self.caller_procs.push(caller_proc);
+
+                let inc_n_data = Symbol::DEV_TMP5;
```

Comment:
Why just start with TMP5 instead of 1?

Reply:
happened over time. will clean up later.